boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -549,6 +549,12 @@ PcMemFinalizeMemoryMap(
     ReserveMemory(MemoryMap, STACKLOW, STACKADDR - STACKLOW, LoaderOsloaderStack, "FreeLdr stack");
     ReserveMemory(MemoryMap, FREELDR_BASE, FrLdrImageSize, LoaderLoadedProgram, "FreeLdr image");
 
+    /* We need to make sure that we don't allocate the 4 MB chunk of physical memory starting at
+     * 0x40000000. The logic used in MempAllocatePTE means that the kernel mapping would be at
```

Comment:
What happens on machines with 1GB of RAM or less?
Does Win7 perhaps expect a non-identity mapping here instead?

Reply:
What has to be known is that the windows NT loaders enable the expected NT paging as soon as they start, and continue running with this paging enabled, while in our case we wait to setup the paging just before loading the NT/reactos kernel. Hence, contrary to the case of windows NT loaders, freeldr has indeed the possibility to write stuff to this physical 0x40000000 address before the expected NT paging has been set up.

Reply:
What I'm saying is that that's not a valid physical address on many machines (namely, those with less than 1GB of RAM).